service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/DraftOrdersController.java

```diff
@@ -71,28 +72,37 @@
     private final HearingBookingService hearingBookingService;
     private final Time time;
     private final RequestData requestData;
+    private final ValidateGroupService validateGroupService;
 
     private static final String JUDGE_AND_LEGAL_ADVISOR_KEY = "judgeAndLegalAdvisor";
 
     @PostMapping("/about-to-start")
     public AboutToStartOrSubmitCallbackResponse handleAboutToStart(@RequestBody CallbackRequest callbackrequest) {
         CaseDetails caseDetails = callbackrequest.getCaseDetails();
+
+        caseDetails.getData().put("dateOfIssue", time.now().toLocalDate());
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(caseDetails.getData())
+            .build();
+    }
+
+    @PostMapping("/date-of-issue/mid-event")
+    public AboutToStartOrSubmitCallbackResponse handleMidEventValidateDateOfIssue(
```

Comment:
I would wrap the entire logic in this method on the condition that there were no errors, no point doing something that is going to have no effect and just be done again